### ISSUE TITLE
Throw expression is not type checked sufficiently

### ISSUE THREAD

**Issue by linusg** [SerenityOS/jakt#147]
This compiles:

```jakt
function foo() throws -> u32 {
    throw 1
}

function main() {
    let x = foo()
    println("{}", x)
}
```

To:

```cpp
#include "runtime/lib.h"


ErrorOr<u32> foo();

ErrorOr<u32> foo(){
using _JaktCurrentFunctionReturnType = ErrorOr<u32>;
{
        return static_cast<i64>(1LL);    }
}

ErrorOr<int> _jakt_main(Array<String>)
{
    using _JaktCurrentFunctionReturnType = ErrorOr<int>;
{
        const u32 x = TRY(foo());
        outln(String("{}"),x);
    }
    return 0;
}
```

Meaning it will unexpectedly print `1`.

**Comment by mattisboeckle** [comment]
Fixed by #166, no longer reproducable

**Comment by linusg** [comment]
nope, still repros, just needs an explicit `return` statement after the `throw`:

```jakt
function foo() throws -> u32 {
    throw 1
    return 2
}

function main() {
    let x = foo()
    println("{}", x)
}
```
